src/functional/fencil_processors/runners/gtfn_cpu.py

```diff
@@ -13,36 +13,61 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 
-from typing import Any
+import dataclasses
+from typing import Any, Final, Optional
 
-import numpy
+import numpy as np
 
-from functional.fencil_processors.builders import cpp as cpp_callable
-from functional.fencil_processors.codegens.gtfn import gtfn_module as gtfn_codegen
-from functional.fencil_processors.processor_interface import fencil_executor
-from functional.iterator import ir
+from functional.fencil_processors import processor_interface as fpi  # fencil processor interface
+from functional.fencil_processors.builders import cache
+from functional.fencil_processors.builders.cpp import bindings, build
+from functional.fencil_processors.codegens.gtfn import gtfn_module
+from functional.fencil_processors.source_modules import cpp_gen, source_modules
+from functional.iterator import ir as itir
 
 
+# TODO(ricoh): Add support for the whole range of arguments that can be passed to a fencil.
 def convert_arg(arg: Any) -> Any:
-    view = numpy.asarray(arg)
+    view = np.asarray(arg)
     if view.ndim > 0:
-        return memoryview(view)
+        return view
     else:
         return arg
```

Comment:
Not critical for this PR, but still suggest to add a ToDo comment here or change it to something more restrictive like:
```python
if hasattr(arg, "__array__"):
    return np.asarray(arg)
else:
    return arg

```

Reply:
I just went ahead and changed it, if it breaks tests I will revert and add a TODO.